src/libs/ajax/data-table-providers/WdsDataTableProvider.ts

```diff
@@ -82,33 +83,61 @@ const getRelationParts = (val: unknown): string[] => {
   return []
 }
 
-// Extract wds URL from Leo response. exported for testing
-export const getWdsUrl = apps => {
-  // look explicitly for an app named 'wds-${app.workspaceId}'. If found, use it, even if it isn't running
-  // this handles the case where the user has explicitly shut down the app
+export const createLeoAppWithErrorHandling = workspaceId => {
+  const typedWithErrorReporting : any = withErrorReporting
+  const createLeoAppCall = typedWithErrorReporting('An error occurred when creating your data tables. Please reach out to support@terra.bio', async () => {
+    await Ajax().Apps.createAppV2(`wds-${workspaceId}`, `${workspaceId}`)
+  })
+  createLeoAppCall()
+}
+
+// Invokes logic to determine a URL for WDS
+// If WDS is not running, a URL will not be present -- in some cases, this function may invoke
+// a new call to Leo to instantiate a WDS being available, thus having a valid URL
+export const resolveWdsUrl = (apps, workspaceId) => {
+  // WDS looks for Kubernetes deployment statuses (such as RUNNING or PROVISIONING), expressed by Leo
+  // See here for specific enumerations -- https://github.com/DataBiosphere/leonardo/blob/develop/core/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/kubernetesModels.scala
+  // look explicitly for a RUNNING app named 'wds-${app.workspaceId}' -- if WDS is healthy and running, there should only be one app RUNNNING
   const namedApp = apps.filter(app => app.appType === 'CROMWELL' && app.appName === `wds-${app.workspaceId}` && app.status === 'RUNNING')
   if (namedApp.length === 1) {
     return namedApp[0].proxyUrls.wds
   }
-  // if we didn't find the expected app 'cbas-wds-default', go hunting:
-  const candidates = apps.filter(app => app.appType === 'CROMWELL' && app.status === 'RUNNING')
+
+  // if we didn't find the expected app 'wds-${app.workspaceId}' running...
+  const candidates = apps.filter(app => app.appType === 'CROMWELL' && app.appName === `wds-${app.workspaceId}`)
+  // ...nothing has launched yet, bring WDS to life!
   if (candidates.length === 0) {
-    // no app deployed yet
+    createLeoAppWithErrorHandling(workspaceId)
     return ''
   }
-  if (candidates.length > 1) {
-    // multiple apps found; use the earliest-created one
-    candidates.sort((a, b) => a.auditInfo.createdDate - b.auditInfo.createdDate)
+
+  // WDS is being created in a Kubernetes cluster (takes a few minutes)
+  if (candidates.length === 1 && candidates[0].status === 'PROVISIONING') {
+    return ''
+  }
+
+  // If we reach this logic, we have more than one Leo app with the associated workspace Id...
+  const allCromwellApps = apps.filter(app => app.appType === 'CROMWELL')
+  if (allCromwellApps.length > 0) {
+    // Evaluate the most-recent-created WDS app
+    allCromwellApps.sort((a, b) => new Date(b.auditInfo.createdDate).valueOf() - new Date(a.auditInfo.createdDate).valueOf())
```

Comment:
this is a notable change in business logic: using the most-recently-created app instead of the earliest-created app. I don't think this is what we want. Imagine this scenario:
* an API user creates a workspace, and manually deploys a WDS app using their own name.
* user adds a bunch of data and does a bunch of work in WDS and has their data tables working the way they want it to.
* user deploys a second WDS app.
* suddenly, the newer WDS app is the only one the UI recognizes, and all of the user's work in their first WDS app is unavailable through the UI. It _looks_ as if all their data disappeared.

I realize this is a corner case, but if we've reached this part of the logic (more than 1 app exists, but no app named `wds-${app.workspaceId}` exists) we're already in a corner case. The motivation for choosing the earliest-created app was to ensure stability for that app.

(in the future, we might want to just add a selector - "choose which app you want to view right now" - but that's pretty far out of scope for this PR)

Reply:
Gotcha -- I think was a miss of mine during refactoring (the logic didn't work prior 😞 ) / interpreting the docstring.

To your point, it would be a slippery slope to start to re-arrange what the end user is seeing -- reverted the `sort` call back and fixed the unit test!